internal/operands/vm-console-proxy/reconcile_test.go

```diff
@@ -201,6 +200,100 @@ var _ = Describe("VM Console Proxy Operand", func() {
 		ExpectResourceNotExists(newRoute(namespace, serviceName), request)
 	})
+
+	Context("with namespace annotation", func() {
+		const otherNamespace = "some-namespace"
+
+		var (
+			serviceAccount *core.ServiceAccount
+			configMap      *core.ConfigMap
+			service        *core.Service
+			deployment     *apps.Deployment
+			route          *routev1.Route
+		)
+
+		BeforeEach(func() {
+			serviceAccount = bundle.ServiceAccount.DeepCopy()
+			serviceAccount.Namespace = otherNamespace
+
+			configMap = bundle.ConfigMap.DeepCopy()
+			configMap.Namespace = otherNamespace
+
+			service = bundle.Service.DeepCopy()
+			service.Namespace = otherNamespace
+
+			deployment = bundle.Deployment.DeepCopy()
+			deployment.Namespace = otherNamespace
+
+			route = newRoute(otherNamespace, serviceName)
+
+			request.Instance.GetAnnotations()[VmConsoleProxyNamespaceAnnotation] = otherNamespace
+		})
+
+		It("should deploy resources in namespace provided by annotation", func() {
+			_, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			ExpectResourceExists(serviceAccount, request)
+			ExpectResourceExists(bundle.ClusterRole, request)
+			ExpectResourceExists(bundle.ClusterRoleBinding, request)
+			ExpectResourceExists(configMap, request)
+			ExpectResourceExists(service, request)
+			ExpectResourceExists(deployment, request)
+			ExpectResourceExists(route, request)
+		})
+
+		It("should cleanup resources from namespace provided by annotation", func() {
+			_, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			ExpectResourceExists(serviceAccount, request)
+			ExpectResourceExists(bundle.ClusterRole, request)
+			ExpectResourceExists(bundle.ClusterRoleBinding, request)
+			ExpectResourceExists(configMap, request)
+			ExpectResourceExists(service, request)
+			ExpectResourceExists(deployment, request)
+			ExpectResourceExists(route, request)
+
+			_, err = operand.Cleanup(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			ExpectResourceNotExists(serviceAccount, request)
+			ExpectResourceNotExists(bundle.ClusterRole, request)
+			ExpectResourceNotExists(bundle.ClusterRoleBinding, request)
+			ExpectResourceNotExists(configMap, request)
+			ExpectResourceNotExists(service, request)
+			ExpectResourceNotExists(deployment, request)
+			ExpectResourceNotExists(route, request)
+		})
+
+		It("should remove resources from the namespace", func() {
+			_, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			ExpectResourceExists(serviceAccount, request)
+			ExpectResourceExists(bundle.ClusterRole, request)
+			ExpectResourceExists(bundle.ClusterRoleBinding, request)
+			ExpectResourceExists(configMap, request)
+			ExpectResourceExists(service, request)
+			ExpectResourceExists(deployment, request)
+			ExpectResourceExists(route, request)
+
+			delete(request.Instance.Annotations, EnableAnnotation)
+			delete(request.Instance.Annotations, VmConsoleProxyNamespaceAnnotation)
```

Comment:
What is the reason of deleting `VmConsoleProxyNamespaceAnnotation` if it's not going to be used in `Reconcile`?

Then without deleting it, it makes sense to write it as: `should remove resources from the namespace once EnableAnnotation deleted`.

Reply:
Both annotations are for the proxy functionality, so I just deleted both.
We can improve it later, this PR is to unblock failing d/s tests.